File: livebox/src/main/java/com/creations/livebox/converters/Converter.java
package com.creations.livebox.converters;

import io.reactivex.annotations.NonNull;

/**
 * @author Sérgio Serra on 26/08/2018.
 * Criations
 * sergioserra99@gmail.com
 */
public interface Converter<R, T> {
    @NonNull
    T convert(R o) throws Exception;
}

File: livebox/src/main/java/com/creations/livebox/converters/Converter.kt
package com.creations.livebox.converters

import io.reactivex.annotations.NonNull

/**
 * @author Sérgio Serra on 26/08/2018.
 * sergioserra99@gmail.com
 */
interface Converter<R, T> {
    @NonNull
    @Throws(Exception::class)
    fun convert(o: R): T
}
